tests/components/esphome/test_config_flow.py

```diff
@@ -48,6 +48,13 @@ def mock_api_connection_error():
         yield mock_error
 
 
+@pytest.fixture(autouse=True)
+def mock_setup_entry():
+    """Mock setting up a config entry."""
+    with patch("homeassistant.components.esphome.async_setup_entry", return_value=True):
```

Comment:
Apparently, the `async_setup_entry` is being called from the config flow (even before the typing change). But now it fails because of a new `is not None` assert due to a mock.

That function shouldn't run in the config flow test anyway, so mock it here.